test/unit/resources/aws_route_tables_test.rb

```diff
@@ -0,0 +1,78 @@
+require 'helper'
+
+class EmptyAwsRouteTablesTest < Minitest::Test
+  def setup
+    AwsRouteTables::BackendFactory.select(AwsMRtbB::Empty)
+  end
+
+  def test_constructor_no_args_ok
+    AwsRouteTables.new
+  end
+
+  def test_constructor_reject_unknown_resource_params
+    assert_raises(ArgumentError) { AwsRouteTables.new(bla: 'blabla') }
+  end
+end
+
+class BasicAwsRouteTablesTest2 < Minitest::Test
+  def setup
+    AwsRouteTables::BackendFactory.select(AwsMRtbB::Basic)
+  end
+
+  def test_search_hit
+    assert AwsRouteTables.new.exists?
+  end
+
+  def test_property_vpc_ids
+    basic = AwsRouteTables.new
+    assert_kind_of(Array, basic.vpc_ids)
+    assert(basic.vpc_ids.include?('vpc-169f777e'))
+    assert(basic.vpc_ids.include?('vpc-169f777d'))
+    refute(basic.vpc_ids.include?(nil))
+  end
+
+  def test_property_route_table_ids
+    basic = AwsRouteTables.new
+    assert_kind_of(Array, basic.route_table_ids)
+    assert(basic.route_table_ids.include?('rtb-2c60ec44'))
+    assert(basic.route_table_ids.include?('rtb-58508630'))
+    refute(basic.route_table_ids.include?(nil))
+  end
+end
+
+# MRtbB = Mock Routetable Backend
+module AwsMRtbB
+  class Empty < AwsBackendBase
+    def describe_route_tables(query)
+      OpenStruct.new(route_tables: [])
+    end
+  end
+
+  class Basic < AwsBackendBase
+    def describe_route_tables(query)
+      fixtures = [
+        OpenStruct.new({associations: [],
+                          propagating_vgws: [],
```

Comment:
I appreciate that this will be useful in the future, but for today, we only test the route_table_id and vpc_id, so it might be best if the other fields were removed for now.